clients/src/main/java/org/apache/kafka/common/record/FileRecords.java

```diff
@@ -47,7 +47,7 @@
 
     // mutable state
     private final AtomicInteger size;
-    private FileChannel channel;
+    private volatile FileChannel channel;
```

Comment:
Not sure why this FileChannel is not volatile in the first place. But after reading the code, it seems the volatile is not necessary. Each thread only access one segment, filerecords, this is centrally allocated through the cleanerManager. There shouldn't be a case where two threads access the same file. If there are thread safety issue, volatile is not enough.

Reply:
@ningliums , I will revisit this tomorrow with you.

Reply:
This was final in the original file. in the last PR final was removed....